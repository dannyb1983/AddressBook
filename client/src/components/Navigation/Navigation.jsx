//modules
import React from "react";
import { Link } from "react-router-dom";
import { Navbar, Container, Nav } from "react-bootstrap";

const Navigation = () => {
	return (
		<Navbar className="sticky-nav" collapseOnSelect expand="lg" bg="dark" variant="dark">
			<Container>
				<Navbar.Brand>
					<i className="fas fa-address-book " />
				</Navbar.Brand>
				<Link to="/" style={{ textDecoration: "none", paddingRight: "50px" }}>
					Address Book
				</Link>
				<Navbar.Toggle aria-controls="responsive-navbar-nav" />
				<Navbar.Collapse id="responsive-navbar-nav">
					<Nav className="ml-auto">
						<Nav.Item as={Link} to="/addcontact" style={{ textDecoration: "none" }}>
								Add New Contact<i className="fas fa-plus ml-2"></i>
						</Nav.Item>
						<Nav.Item>
						</Nav.Item>
					</Nav>
				</Navbar.Collapse>
			</Container>
		</Navbar>
	);
};

export default Navigation;
