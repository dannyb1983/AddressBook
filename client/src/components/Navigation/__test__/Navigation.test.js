//modules
import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
// Components
import Navigation from '../Navigation';

describe('Navigation (Unit)', () => {
  afterEach(() => cleanup());
  const renderNavigation = () => render(<MemoryRouter initialEntries={['/']}><Navigation /></MemoryRouter>)
  const renderEmptyNavigation = () => render(<MemoryRouter><Navigation /></MemoryRouter>)

  describe('render Navigation()', () => {
    const { asFragment } = renderNavigation();

    it('Navigation render should match snapshot', () => {
      renderNavigation();
      expect(asFragment()).toMatchSnapshot();
    });
    it('All links should render to the Nav', () => {
      renderNavigation();
      expect(screen.getByText('Address Book')).toBeInTheDocument();
      expect(screen.getByText('Add New Contact')).toBeInTheDocument();
    });
  });
});
