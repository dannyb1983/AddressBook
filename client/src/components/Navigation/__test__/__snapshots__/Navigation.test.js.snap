// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation (Unit) render Navigation() Navigation render should match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="sticky-nav navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
      >
        <i
          class="fas fa-address-book "
        />
      </span>
      <a
        href="/"
        style="text-decoration: none; padding-right: 50px;"
      >
        Address Book
      </a>
      <button
        aria-controls="responsive-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="responsive-navbar-nav"
      >
        <div
          class="ml-auto navbar-nav"
        >
          <a
            class="nav-item"
            href="/addcontact"
            style="text-decoration: none;"
          >
            Add New Contact
            <i
              class="fas fa-plus ml-2"
            />
          </a>
          <div
            class="nav-item"
          />
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
